wear/src/main/java/io/homeassistant/companion/android/home/views/MainView.kt

```diff
@@ -64,11 +64,11 @@ fun MainView(
     var expandedScripts: Boolean by rememberSaveable { mutableStateOf(true) }
     var expandedSwitches: Boolean by rememberSaveable { mutableStateOf(true) }
 
-    val scenes = entities.filter { it.key.split(".")[0] == "scene" }.values.toList()
-    val scripts = entities.filter { it.key.split(".")[0] == "script" }.values.toList()
-    val lights = entities.filter { it.key.split(".")[0] == "light" }.values.toList()
-    val inputBooleans = entities.filter { it.key.split(".")[0] == "input_boolean" }.values.toList()
-    val switches = entities.filter { it.key.split(".")[0] == "switch" }.values.toList()
+    val scenes = entities.filter { it.key.split(".")[0] == "scene" }.values.toList().sortedBy { it.entityId }
+    val scripts = entities.filter { it.key.split(".")[0] == "script" }.values.toList().sortedBy { it.entityId }
+    val lights = entities.filter { it.key.split(".")[0] == "light" }.values.toList().sortedBy { it.entityId }
+    val inputBooleans = entities.filter { it.key.split(".")[0] == "input_boolean" }.values.toList().sortedBy { it.entityId }
+    val switches = entities.filter { it.key.split(".")[0] == "switch" }.values.toList().sortedBy { it.entityId }
```

Comment:
We make a temporary list of entities, sort it, then filter the sorted list.

```suggestion
    val entitieList = entities.values.toList().sortedBy { it.entityId }
    val scenes = entitieList.filter { it.key.split(".")[0] == "scene" }
    val scripts = entitieList.filter { it.key.split(".")[0] == "script" }
    val lights = entitieList.filter { it.key.split(".")[0] == "light" }
    val inputBooleans = entitieList.filter { it.key.split(".")[0] == "input_boolean" }
    val switches = entitieList.filter { it.key.split(".")[0] == "switch" }
```

Reply:
you were close :) had to replace `key` with `entityId` as we were already using the values to sort by

Reply:
It was from my phone too, cut me some slack, haha

Reply:
hahahha for sure :P